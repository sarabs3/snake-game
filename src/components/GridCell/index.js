import React from "react";

const GridCell = props => {
	const classes = `grid-cell 
		${props.foodCell ? "grid-cell--food" : ""} 
		${props.snakeCell ? "grid-cell--snake" : ""}
		`;

	return (
		<div
			className={classes}
			style={{height: props.size + "px", width: props.size + "px"}}
			data-testid={'grid-cell'}
		/>
	);
};

export default GridCell;
