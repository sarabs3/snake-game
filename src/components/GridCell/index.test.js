import React from 'react';
import ReactDOM from 'react-dom';

import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';

import GridCell from './index';

let div;

beforeAll(() => {
	div = document.createElement('div');
	document.body.appendChild(div);
});

afterAll(() => {
	ReactDOM.unmountComponentAtNode(div);
	div.remove();
	div = null;
});

it('<GridCell /> renders without crashing', () => {
	const {queryByTestId} = render(<GridCell foodCell={true} snakeCell={false} size={15}/>, div);

	expect(queryByTestId('grid-cell')).toBeTruthy();
});

describe('Grid cell snapshots', () => {
	it('<GridCell /> food cell matches snapshot', () => {
		const tree = renderer.create(<GridCell foodCell={true} snakeCell={false} size={15} />, div);

		expect(tree).toMatchSnapshot();
	});

	it('<GridCell /> snake cell matches snapshot', () => {
		const tree = renderer.create(<GridCell foodCell={false} snakeCell={true} size={15} />, div);

		expect(tree).toMatchSnapshot();
	});
});
