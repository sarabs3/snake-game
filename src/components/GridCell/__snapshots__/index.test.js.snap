// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid cell snapshots <GridCell /> food cell matches snapshot 1`] = `
<div
  className="grid-cell 
		grid-cell--food 
		
		"
  data-testid="grid-cell"
  style={
    Object {
      "height": "15px",
      "width": "15px",
    }
  }
/>
`;

exports[`Grid cell snapshots <GridCell /> snake cell matches snapshot 1`] = `
<div
  className="grid-cell 
		 
		grid-cell--snake
		"
  data-testid="grid-cell"
  style={
    Object {
      "height": "15px",
      "width": "15px",
    }
  }
/>
`;
